DS-Session1-Question1/src/utility/LinkedList.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * @author Abhishek Bishnoi
+ * @since Jul 29, 2019
+ * This Class Provides Dynamic List as where,
+ * it provides various methods like add, toString, substring, rotateSubstring etc etc
+ */
+package utility;
+
+public class LinkedList {
+	Node headNode=null;
+	Node lastNode=null;
+	/**
+	 * This Function is used to Add Data into Node And Connecting them with Next Node 
+	 * @param value : value to adding in node
+	 * @return boolean: true or false for operation success or not
+	 * @exception : when adding of element can't be done
+	 */
+	public boolean add(Node node){
+		if(node==null){
+			throw new AssertionError("Node Can't Be Null");
+		}
+		try{
+			if(lastNode==null){
+				headNode=node;
+				lastNode=node;
+			}else{
+				lastNode.next = node;
+				lastNode = node;
+			}
+			return true;
+		}catch(Exception exception){
+			throw new AssertionError("Error in Insertion");
+		}
+	}
+	/**
+	 * This Method is used to rotate SubList from start to end index
+	 * @param start: start index
+	 * @param end: end index
+	 * @exception AssertionError: when we get invalid start index, end index
+	 */
+	public void rotateSubList(int start, int end, int rotateIndex){
```

Comment:
We can break this method into 3 parts:
1. get subList from a linked list.
2. rotate the sublist.
3. insert the sublist into the linked list.

I think, it would be more readable and well distributed ?